### ISSUE TITLE
[TTIR→TTNN] Floating-point exception when lowering ops with zero-size tensor dimensions in pocket_tts model

### ISSUE THREAD

**Issue by ashokkumarkannan1** [tenstorrent/tt-mlir#7360]
### Describe the bug

- The TTIR-to-TTNN lowering pass crashes with a fatal floating-point exception when processing graphs that contain tensors with a zero-size dimension (e.g., tensor<1x0x32xf32>).
- This was discovered while running the `pocket_tts` model. The `FlowLMModel.forward()` method passes a `[1, 0, 32] float32 tensor` through `torch.where `and `nn.Linear.`
- PyTorch handles zero-size tensors correctly, but when this graph is lowered from TTIR to TTNN, the compiler hits a floating-point exception
- The TTIR module is generated successfully and printed in full. The crash occurs immediately after, before any TTNN output is produced.

- TTIR graph that triggers the crash:
```
module @SyncTensorsGraph.12 attributes {mhlo.cross_program_prefetches = [], mhlo.input_output_alias = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false, ttcore.meshes = #ttcore.meshes<[<"mesh" = 1x1>]>} {
  ttcore.device_module {
    builtin.module @SyncTensorsGraph.12 attributes {mhlo.cross_program_prefetches = [], mhlo.input_output_alias = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false, ttcore.meshes = #ttcore.meshes<[<"mesh" = 1x1>]>} {
      func.func @main(%arg0: tensor<1x0x32xf32> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<1x0x32xf32>>} loc("p0.1"), %arg1: tensor<32xf32> {ttcore.argument_type = #ttcore.argument_type<input>, ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<32xf32>>} loc("p1.3")) -> (tensor<1x0x32xf32> {ttcore.runtime_tensor_sharding = #ttcore<runtime_tensor_sharding shard_status = <unsharded>, local_shape = tensor<1x0x32xf32>>}) {
        %0 = "ttir.ne"(%arg0, %arg0) : (tensor<1x0x32xf32>, tensor<1x0x32xf32>) -> tensor<1x0x32xi1> loc(#loc3)
        %1 = "ttir.reshape"(%arg1) <{shape = [1 : i32, 32 : i32]}> : (tensor<32xf32>) -> tensor<1x32xf32> loc(#loc4)
        %2 = "ttir.reshape"(%1) <{shape = [1 : i32, 1 : i32, 32 : i32]}> : (tensor<1x32xf32>) -> tensor<1x1x32xf32> loc(#loc5)
        %3 = "ttir.broadcast"(%2) <{broadcast_dimensions = array<i64: 1, 0, 1>}> : (tensor<1x1x32xf32>) -> tensor<1x0x32xf32> loc(#loc5)
        %4 = "ttir.where"(%0, %3, %arg0) : (tensor<1x0x32xi1>, tensor<1x0x32xf32>, tensor<1x0x32xf32>) -> tensor<1x0x32xf32> loc(#loc6)
        return %4 : tensor<1x0x32xf32> loc(#loc)
      } loc(#loc)
    } loc(#loc)
  } loc(#loc)
} loc(#loc)
```

### Steps to reproduce the issue ([zero_dim_sanity.log](https://github.com/user-attachments/files/25768079/zero_dim_sanity.log))

```
git checkout akannan/pocket_tts_fpe
# Use the standalone repro script at test_where_zero_dim_repro.py
pytest -svv test_where_zero_dim_repro.py
# contains sanities for `torch.where(torch.isnan(x), bos_emb, x)` and `torch.where + nn.Linear(32, 1024) fused` 
```

Impacting Environment:
Model: pocket_tts (FlowLMModel.forward) (whole_model log - [whole_model_pocket_tts.log](https://github.com/user-attachments/files/25768245/whole_model_pocket_tts.log))

Shapes: sequence=[1, 0, 32] f32, bos_emb=[32] f32, input_linear.weight=[1024, 32] f32

**Comment by ashokkumarkannan1** [comment]
@sdjordjevicTT Could you please point me to the owner who can help with this issue ? thanks!

**Comment by sdjordjevicTT** [comment]
@ashokkumarkannan1, is this the same issue as here?
https://github.com/tenstorrent/tt-mlir/issues/6786

**Comment by sdjordjevicTT** [comment]
@mrakitaTT this would be all folded out if we have enabled this stablehlo pass:
https://github.com/tenstorrent/tt-mlir/issues/5671

**Comment by ashokkumarkannan1** [comment]
> is this the same issue as here?

They are related but not the same issue. Both are triggered by zero-size tensor dimensions.
#6786: torch.cat op fails with an MLIR assertion (collapsedLinearAffineMap: "Dim does not participate in AffineMap RHS")
#7360: torch.where op fails with a fatal floating-point exception during TTIR→TTNN lowering

**Comment by sdjordjevicTT** [comment]
@mrakitaTT is there a way that we can prioritize the stable hlo aggressive simplification work?

**Comment by mrakitaTT** [comment]
@sdjordjevicTT sure, @ajakovljevicTT will sync with @anuragsingh-tt to unblock this
